uniffi_bindgen/src/bindings/kotlin/templates/CustomTypeTemplate.kt

```diff
@@ -1,41 +1,42 @@
 {%- match config %}
 {%- when None %}
-{#- No custom type config, just forward all methods to our builtin type #}
-public typealias {{ outer|ffi_converter_name }} = {{ builtin|ffi_converter_name }}
-
-{#- Create a typealias so that this type can be imported when used as an external type #}
+{#- Define the type using typealiases to the builtin #}
 public typealias {{ name }} = {{ builtin|type_name }}
+public typealias {{ outer|ffi_converter_name }} = {{ builtin|ffi_converter_name }}
 
 {%- when Some with (config) %}
-{%- let type_name=self.type_name(config) %}
-{%- let ffi_type_name=self.builtin_ffi_type().borrow()|ffi_type_name %}
 
-{#- Create a typealias so that this type can be imported when used as an external type #}
-public typealias {{ name }} = {{ type_name }}
+{%- let ffi_type_name=builtin.ffi_type().borrow()|ffi_type_name %}
+
+{# When the config specifies a different type name, create a typealias for it #}
+{%- match config.type_name %}
+{%- when Some(concrete_type_name) %}
+public typealias {{ name }} = {{ concrete_type_name }}
+{%- else %}
+{%- endmatch %}
```

Comment:
… and this is the bit we declare

```kt
public typealias JsonObject = JSONObject
```

I can't tell if this is going to be a breaking change or not in all the languages. But: it looks like APIs that we written `JSONObject` now get the less friendly `JsonObject`.

If so, you should probably update the docs, and possibly add a **⚠️ Breaking Change ⚠️** for all relevant languages.

I'd also add a comment to the `typealias` that IDEs can pick up so foreign language developers understand what and why this is.

Aesthetically, I don't really like this, so am hoping we can use the old behaviour, but I'm willing to have a bit of technical debt to keep your momentum going.

So be aware, all who use it, this is a breaking change.

Reply:
Agreed that this is the biggest downside.  If I understand how typealiases work, the consumer code can stay the same, but this could make the generated docs and code completion worse. 

I like the idea of a comment to try to mitigate this.

If we wanted to keep using `JSONObject` in the Kotlin function signatures, I think there are some options.  Maybe the simplest would be adding a `concrete_type_name` field to `Type::Custom`.  It would start blank then the bindings code could set it.